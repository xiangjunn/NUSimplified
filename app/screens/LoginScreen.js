import React from 'react'
import { Container, Form} from 'native-base';
import { StyleSheet, Image} from 'react-native';

import LoginDetails from '../container/LoginAuth';


function LoginScreen() {
    return (
        <Container>
          {/** Upper half of the screen, to display logo */}
            <Form style={{flex: 1, alignItems: "center", justifyContent: "center"}}>
                <Image style={styles.logo} source={require("../assets/logo.png")}></Image>
            </Form>

          {/** Bottom half of the screen, to display login info */}
            <Form style={{flex: 2}}>
              <LoginDetails></LoginDetails>
            </Form>
          </Container>
        
        
        );
  }

  const styles = StyleSheet.create({
    logo: { // for our app logo
        resizeMode: 'cover',
        width: '100%',
        height: '100%'
    }
  })


  export default LoginScreen;