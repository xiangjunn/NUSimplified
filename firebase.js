import * as firebase from 'firebase';
import '@firebase/auth';
import '@firebase/firestore';
import {API_KEY, PROJECT_NAME, MESSAGING_SENDER_ID, APP_ID} from '@env'

const firebaseConfig = {
  apiKey: API_KEY,
  authDomain: PROJECT_NAME + '.firebaseapp.com',
  databaseURL: 'https://' + PROJECT_NAME +'.firebaseio.com',
  projectId: PROJECT_NAME,
  storageBucket: PROJECT_NAME + '.appspot.com',
  messagingSenderId: MESSAGING_SENDER_ID,
  appId: APP_ID,
};

if (!firebase.apps.length) {
    firebase.initializeApp(firebaseConfig);
}

export { firebase };