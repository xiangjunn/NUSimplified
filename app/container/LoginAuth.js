import React, { useState } from 'react'
import { useNavigation } from '@react-navigation/native';
import { firebase } from '../../firebase'
import { Content, Form, Item, Input, Icon, Button, Text, Label} from 'native-base';
import { StyleSheet, Modal, TouchableOpacity, Alert, ActivityIndicator } from 'react-native';

function LoginAuth() {
    const navigation = useNavigation();
    const [email, setEmail] = useState('')
    const [emailSelected, setEmailSelected] = useState(false)
    const [password, setPassword] = useState('')
    const [passwordSelected, setPasswordSelected] = useState(false)
    const [modalVisible, setModalVisible] = useState(false);
    const [forgotPasswordEmail, setForgotPasswordEmail] = useState('');
    const [loading, setLoading] = useState(false);

    const onLoginPress = () => {
      setLoading(true);
      firebase
          .auth()
          .signInWithEmailAndPassword(email.toLowerCase().trim(), password)
          .then((response) => {
              const uid = response.user.uid
              const usersRef = firebase.firestore().collection('users')
              usersRef
                  .doc(uid)
                  .get()
                  .then(firestoreDocument => {
                      if (!firestoreDocument.exists) {
                          alert("User does not exist anymore.")
                          setLoading(false);
                          return;
                      } else if (!firebase.auth().currentUser.emailVerified) {
                          alert("Email not verified!")
                          setLoading(false);
                          firebase.auth().signOut()
                      }
                  })
                  .catch(error => {
                      alert(error)
                      setLoading(false);
                  });
          })
          .catch(error => {
              alert(error)
              setLoading(false);
          })
    }

    function resetPassword() {
        firebase.auth().sendPasswordResetEmail(forgotPasswordEmail.toLowerCase().trim())
        .then(() => {
          // Password reset email sent!
          Alert.alert("A password reset email has been sent. Please check your mailbox.");
          setModalVisible(!modalVisible);
        })
        .catch((error) => {
          alert(error);
        });
    }

    return (
        <Content style={styles.login}>
            <Form>
              {/** Email textbox */}
              <Item rounded last style={emailSelected ? {borderColor: "#00BBBB"} : {}}>
                <Icon type='FontAwesome5' name='envelope' style={emailSelected ? {color: "#00BBBB"} : {}}/>
                <Input
                  keyboardType='email-address'
                  placeholder='Email'
                  underlineColorAndroid="transparent"
                  onChangeText={(text) => setEmail(text)}
                  value={email}
                  onFocus={() => setEmailSelected(true)}
                  onBlur={() => setEmailSelected(false)}
                >
                </Input>
              </Item>

              {/** Password textbox */}
              <Item rounded last style={passwordSelected ? {borderColor: "#00BBBB"} : {}}>
                <Icon type='FontAwesome5' name='key' style={passwordSelected ? {color: "#00BBBB"} : {}}/>
                <Input
                  placeholder='Password'
                  secureTextEntry={true}
                  onChangeText={(text) => setPassword(text)}
                  value={password}
                  onFocus={() => setPasswordSelected(true)}
                  onBlur={() => setPasswordSelected(false)}
                   ></Input>
              </Item>

              {/** "Forget password" button */}
              <TouchableOpacity style={{height: 35, marginTop: 5}} onPress={() => setModalVisible(!modalVisible)}>
                  <Label style={{color: '#0645AD', marginLeft: 10}}>Forgot password?</Label>
              </TouchableOpacity>
            
              {/** Login button */}
              <Button rounded success style={styles.margin} onPress={() => onLoginPress()}>
              {loading ? <ActivityIndicator animating={true} size="large" style={{flex: 1, alignSelf: 'center'}} color="#999999" />
                :  <Text style={styles.textAlign} >Login</Text>
              }
              </Button>
                
              {/** Signup button */}
              <Button rounded style={styles.margin} onPress={() => navigation.replace("Signup")}>
                  <Text style={styles.textAlign} >Sign up</Text>
              </Button>
            </Form>

            <Modal
        animationType="slide"
        transparent={true}
        visible={modalVisible}
        onRequestClose={() => {
          setModalVisible(!modalVisible);
        }}
      >
        <Form style={styles.centeredView}>
          <Form style={styles.modalView}>
            <Text style={styles.modalText}>Please input your email address for password reset</Text>
            <Item regular last>
                <Input
                  placeholder='eg e0123456@u.nus.edu'
                  onChangeText={(text) => setForgotPasswordEmail(text)}
                  value={forgotPasswordEmail}
                  keyboardType='number-pad'
                >
                </Input>
                  
            </Item>
            <Form style={{flexDirection: 'row', marginTop: 25}}>
                  <Button danger style={styles.button} onPress={() => {
                      setModalVisible(!modalVisible)
                      setForgotPasswordEmail('')
                      }}>
                      <Text style={styles.text}>Cancel</Text>
                  </Button>
                  <Button success style={styles.button} onPress={() => resetPassword()}>
                      <Text style={[styles.text, {color: 'black'}]}>Confirm</Text>
                  </Button>
                  </Form>
          </Form>
        </Form>
      </Modal>
        </Content>
      );
  }

  const styles = StyleSheet.create({
      login: { // for email and password
          marginLeft: '5%',
          marginRight: '5%',
          alignContent: 'center',
      },
      margin: { // for login and signup button
        marginTop: 40,
        height: 60,
        flex: 1
      },
      textAlign: {
          flex: 1,
          textAlign: 'center'
      },
      centeredView: {
        flex: 1,
        justifyContent: "center",
        alignItems: "center",
        marginTop: 22
      },
      modalView: {
        margin: 20,
        backgroundColor: "white",
        borderRadius: 20,
        padding: 35,
        alignItems: "center",
        shadowColor: "#000",
        shadowOffset: {
          width: 0,
          height: 2
        },
        shadowOpacity: 0.25,
        shadowRadius: 4,
        elevation: 5
      },
      modalText: {
        marginBottom: 15,
        textAlign: "center",
        fontWeight: "bold"
      },
      text: {
        flex: 1,
        textAlign: 'center'
    },
    button: {
        flex: 1,
        alignSelf: 'center',
        marginHorizontal: '5%',
        borderRadius: 50
    },
    selected: {

    },
    notSelected: {

    }
  })

  export default LoginAuth;