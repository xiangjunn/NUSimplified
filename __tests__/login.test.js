import React from 'react';
import { fireEvent, render, waitFor, cleanup } from '@testing-library/react-native';
import { Alert } from 'react-native';
import LoginScreen from '../app/screens/LoginScreen';
import LoginAuth from '../app/container/LoginAuth';
import { SAMPLE_EMAIL, SAMPLE_PASSWORD, SAMPLE_EMAIL_UNVERIFIED, SAMPLE_PASSWORD_UNVERIFIED } from '@env'
import { firebase } from '../firebase'

afterEach(() => {    
  jest.clearAllMocks();
  cleanup();
});

const mockReplace = jest.fn();
jest.mock("@react-navigation/native", () => {
    const actualNav = jest.requireActual("@react-navigation/native");
    return {
      ...actualNav,
      useNavigation: () => ({
        navigate: jest.fn(),
        replace: mockReplace,
        dispatch: jest.fn(),
      }),
    };
});

window.addEventListener = jest.fn();
window.attachEvent= jest.fn();

describe('Login Test', () => {
    test('renders correctly', () => {
        const tree = render(<LoginScreen />).toJSON();
        expect(tree).toMatchSnapshot();
    });  

    test('dialog pops up after clicking forgot password', () => {
        const { getByTestId } = render(<LoginAuth />);
        const button = getByTestId('forgotPassword');
        expect(getByTestId('modal').props.visible).toBe(false);
        fireEvent.press(button);
        expect(getByTestId('modal').props.visible).toBe(true);
    })

    test('navigate to registration screen upon pressing sign up button', () => {
      const { getByTestId } = render(<LoginAuth />);
      const button = getByTestId('signUp');
      fireEvent.press(button);
      expect(mockReplace).toBeCalled();
      expect(mockReplace).toBeCalledWith('Signup');
    })

    test('login with empty email field', async () => {
      const email = '';
      const password = '12345678';
      const { getByTestId, getByPlaceholderText } = render(<LoginAuth />);
      fireEvent.changeText(getByPlaceholderText('Email'), email);
      fireEvent.changeText(getByPlaceholderText('Password'), password);
      const button = getByTestId('login');
      const spy = jest.spyOn(Alert, 'alert');
      fireEvent.press(button)
      await waitFor(() => expect(spy).toBeCalledWith('Email or password is not filled in'))
    })

    test('login with empty password field', async () => {
      const email = SAMPLE_EMAIL;
      const password = '';
      const { getByTestId, getByPlaceholderText } = render(<LoginAuth />);
      fireEvent.changeText(getByPlaceholderText('Email'), email);
      fireEvent.changeText(getByPlaceholderText('Password'), password);
      const button = getByTestId('login');
      const spy = jest.spyOn(Alert, 'alert');
      fireEvent.press(button)
      await waitFor(() => expect(spy).toBeCalledWith('Email or password is not filled in'))
    })

    test('login with non-NUS email', async () => {
      const email = 'notnusemail@gmail.com';
      const password = '12345678';
      const { getByTestId, getByPlaceholderText } = render(<LoginAuth />);
      fireEvent.changeText(getByPlaceholderText('Email'), email);
      fireEvent.changeText(getByPlaceholderText('Password'), password);
      const button = getByTestId('login');
      const spy = jest.spyOn(Alert, 'alert');
      fireEvent.press(button)
      await waitFor(() => expect(spy).toBeCalledWith('This is not a NUS email'))
    })

    test('login with an email that is not registered yet', async () => {
      const email = 'e9999999@u.nus.edu';
      const password = '12345678';
      const { getByTestId, getByPlaceholderText } = render(<LoginAuth />);
      fireEvent.changeText(getByPlaceholderText('Email'), email);
      fireEvent.changeText(getByPlaceholderText('Password'), password);
      const button = getByTestId('login');
      const spy = jest.spyOn(Alert, 'alert');
      fireEvent.press(button)
      await waitFor(() => expect(spy).toBeCalledWith('Email is not registered. Please register an account.'))
    })

    test('login with correct email but wrong password', async () => {
      const email = SAMPLE_EMAIL;
      const password = 'qwertyu';
      const { getByTestId, getByPlaceholderText } = render(<LoginAuth />);
      fireEvent.changeText(getByPlaceholderText('Email'), email);
      fireEvent.changeText(getByPlaceholderText('Password'), password);
      const button = getByTestId('login');
      const spy = jest.spyOn(Alert, 'alert');
      fireEvent.press(button)
      await waitFor(() => expect(spy).toBeCalledWith('The password is invalid'))
    })

    test('login with correct login credentials but email not verified', async () => {
      const email = SAMPLE_EMAIL_UNVERIFIED;
      const password = SAMPLE_PASSWORD_UNVERIFIED.toString();
      const { getByTestId, getByPlaceholderText } = render(<LoginAuth />);
      fireEvent.changeText(getByPlaceholderText('Email'), email);
      fireEvent.changeText(getByPlaceholderText('Password'), password);
      const button = getByTestId('login');
      const spy = jest.spyOn(Alert, 'alert');
      fireEvent.press(button)
      await waitFor(() => expect(spy).toBeCalledWith('Email not verified!'))
    })

    test('login with correct login credentials and email verified', async () => {
      const email = SAMPLE_EMAIL;
      const password = SAMPLE_PASSWORD.toString();
      let loggedIn = false;
      const { getByTestId, getByPlaceholderText } = render(<LoginAuth />);
      fireEvent.changeText(getByPlaceholderText('Email'), email);
      fireEvent.changeText(getByPlaceholderText('Password'), password);
      const button = getByTestId('login');
      const spy = jest.spyOn(Alert, 'alert');
      fireEvent.press(button)
      firebase.auth().onAuthStateChanged(function(user) {
        if (user) {
          // User is logged in.
          loggedIn = true;
        } else {
          loggedIn = false;
        }
      });
      await waitFor(() => {
        expect(spy).not.toBeCalled();
        expect(loggedIn).toBe(false);
      })
    })
});
