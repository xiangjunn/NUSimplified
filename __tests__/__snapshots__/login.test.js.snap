// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login Test renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "flex": 1,
      "height": 1334,
    }
  }
>
  <View
    style={
      Array [
        Object {},
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
        },
      ]
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": "100%",
          "resizeMode": "cover",
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 2,
        },
      ]
    }
  >
    <RCTSafeAreaView
      emulateUnlessSupported={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
    >
      <RCTScrollView
        automaticallyAdjustContentInsets={false}
        contentContainerStyle={
          Array [
            Object {
              "padding": undefined,
            },
            undefined,
          ]
        }
        contentInset={
          Object {
            "bottom": 0,
          }
        }
        enableAutomaticScroll={true}
        enableOnAndroid={false}
        enableResetScrollToCoords={true}
        extraHeight={75}
        extraScrollHeight={0}
        getScrollResponder={[Function]}
        handleOnScroll={[Function]}
        keyboardDismissMode="interactive"
        keyboardOpeningTime={250}
        keyboardShouldPersistTaps="handled"
        keyboardSpace={0}
        onScroll={[Function]}
        resetKeyboardSpace={[Function]}
        resetScrollToCoords={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        scrollEventThrottle={1}
        scrollForExtraHeightOnAndroid={[Function]}
        scrollIntoView={[Function]}
        scrollToEnd={[Function]}
        scrollToFocusedInput={[Function]}
        scrollToPosition={[Function]}
        showsVerticalScrollIndicator={true}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            },
            Object {
              "alignContent": "center",
              "marginLeft": "5%",
              "marginRight": "5%",
            },
          ]
        }
        update={[Function]}
        viewIsInsideTabBar={false}
      >
        <View>
          <View
            style={Object {}}
          >
            <View
              accessible={true}
              focusable={false}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 1,
                  "borderColor": "#D9D5DC",
                  "borderLeftWidth": 1,
                  "borderRadius": 30,
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginLeft": 0,
                  "opacity": 1,
                  "paddingLeft": 15,
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                editable={true}
                keyboardType="email-address"
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Email"
                placeholderTextColor="#575757"
                rejectResponderTermination={true}
                style={
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 8,
                    "paddingRight": 5,
                    "top": 1.5,
                  }
                }
                underlineColorAndroid="transparent"
                value=""
              />
            </View>
            <View
              accessible={true}
              focusable={false}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderBottomWidth": 1,
                  "borderColor": "#D9D5DC",
                  "borderLeftWidth": 1,
                  "borderRadius": 30,
                  "borderRightWidth": 1,
                  "borderTopWidth": 1,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "marginLeft": 0,
                  "opacity": 1,
                  "paddingLeft": 15,
                }
              }
            >
              <Text />
              <TextInput
                allowFontScaling={true}
                editable={true}
                onBlur={[Function]}
                onChangeText={[Function]}
                onFocus={[Function]}
                placeholder="Password"
                placeholderTextColor="#575757"
                rejectResponderTermination={true}
                secureTextEntry={true}
                style={
                  Object {
                    "color": "#000",
                    "flex": 1,
                    "fontSize": 17,
                    "height": 50,
                    "paddingLeft": 8,
                    "paddingRight": 5,
                    "top": 1.5,
                  }
                }
                underlineColorAndroid="rgba(0,0,0,0)"
                value=""
              />
            </View>
            <View
              accessible={true}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "height": 35,
                  "marginTop": 5,
                  "opacity": 1,
                }
              }
              testID="forgotPassword"
            >
              <Text
                style={
                  Array [
                    Object {
                      "fontSize": 17,
                    },
                    Object {
                      "color": "#0645AD",
                      "marginLeft": 10,
                    },
                  ]
                }
              >
                Forgot password?
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#5cb85c",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 57,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 60,
                  "justifyContent": "space-between",
                  "marginTop": 40,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
              testID="login"
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 16.5,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                    },
                    Object {
                      "flex": 1,
                      "textAlign": "center",
                    },
                  ]
                }
                uppercase={false}
              >
                Login
              </Text>
            </View>
            <View
              accessible={true}
              focusable={true}
              forwardedRef={[Function]}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignItems": "center",
                  "alignSelf": "flex-start",
                  "backgroundColor": "#007aff",
                  "borderBottomWidth": null,
                  "borderColor": "#007aff",
                  "borderLeftWidth": null,
                  "borderRadius": 57,
                  "borderRightWidth": null,
                  "borderTopWidth": null,
                  "borderWidth": undefined,
                  "elevation": 2,
                  "flex": 1,
                  "flexDirection": "row",
                  "height": 60,
                  "justifyContent": "space-between",
                  "marginTop": 40,
                  "opacity": 1,
                  "paddingBottom": 6,
                  "paddingTop": 6,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                }
              }
              testID="signUp"
            >
              <Text
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "#fff",
                      "fontFamily": "System",
                      "fontSize": 16.5,
                      "marginLeft": 0,
                      "marginRight": 0,
                      "paddingLeft": 16,
                      "paddingRight": 16,
                    },
                    Object {
                      "flex": 1,
                      "textAlign": "center",
                    },
                  ]
                }
                uppercase={false}
              >
                Sign up
              </Text>
            </View>
          </View>
          <Modal
            animationType="slide"
            hardwareAccelerated={false}
            onRequestClose={[Function]}
            testID="modal"
            transparent={true}
            visible={false}
          >
            <View
              style={
                Array [
                  Object {},
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "justifyContent": "center",
                    "marginTop": 22,
                  },
                ]
              }
            >
              <View
                style={
                  Array [
                    Object {},
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "white",
                      "borderRadius": 20,
                      "elevation": 5,
                      "margin": 20,
                      "padding": 35,
                      "shadowColor": "#000",
                      "shadowOffset": Object {
                        "height": 2,
                        "width": 0,
                      },
                      "shadowOpacity": 0.25,
                      "shadowRadius": 4,
                    },
                  ]
                }
              >
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#000",
                        "fontFamily": "System",
                        "fontSize": 16,
                      },
                      Object {
                        "fontWeight": "bold",
                        "marginBottom": 15,
                        "textAlign": "center",
                      },
                    ]
                  }
                  uppercase={false}
                >
                  Please input your email address for password reset
                </Text>
                <View
                  accessible={true}
                  focusable={false}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "backgroundColor": "transparent",
                      "borderBottomWidth": 1,
                      "borderColor": "#D9D5DC",
                      "borderLeftWidth": 1,
                      "borderRadius": undefined,
                      "borderRightWidth": 1,
                      "borderTopWidth": 1,
                      "borderWidth": undefined,
                      "flexDirection": "row",
                      "marginLeft": 0,
                      "opacity": 1,
                      "paddingLeft": 15,
                    }
                  }
                >
                  <TextInput
                    allowFontScaling={true}
                    editable={true}
                    keyboardType="number-pad"
                    onChangeText={[Function]}
                    placeholder="eg e0123456@u.nus.edu"
                    placeholderTextColor="#575757"
                    rejectResponderTermination={true}
                    style={
                      Object {
                        "color": "#000",
                        "flex": 1,
                        "fontSize": 17,
                        "height": 50,
                        "paddingLeft": 8,
                        "paddingRight": 5,
                        "top": 1.5,
                      }
                    }
                    underlineColorAndroid="rgba(0,0,0,0)"
                    value=""
                  />
                </View>
                <View
                  style={
                    Array [
                      Object {},
                      Object {
                        "flexDirection": "row",
                        "marginTop": 25,
                      },
                    ]
                  }
                >
                  <View
                    accessible={true}
                    focusable={true}
                    forwardedRef={[Function]}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "backgroundColor": "#d9534f",
                        "borderBottomWidth": null,
                        "borderColor": "#007aff",
                        "borderLeftWidth": null,
                        "borderRadius": 50,
                        "borderRightWidth": null,
                        "borderTopWidth": null,
                        "borderWidth": undefined,
                        "elevation": 2,
                        "flex": 1,
                        "flexDirection": "row",
                        "height": 45,
                        "justifyContent": "space-between",
                        "marginHorizontal": "5%",
                        "opacity": 1,
                        "paddingBottom": 6,
                        "paddingTop": 6,
                        "shadowColor": undefined,
                        "shadowOffset": undefined,
                        "shadowOpacity": undefined,
                        "shadowRadius": undefined,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#fff",
                            "fontFamily": "System",
                            "fontSize": 16.5,
                            "marginLeft": 0,
                            "marginRight": 0,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                          },
                          Object {
                            "flex": 1,
                            "textAlign": "center",
                          },
                        ]
                      }
                      uppercase={false}
                    >
                      Cancel
                    </Text>
                  </View>
                  <View
                    accessible={true}
                    focusable={true}
                    forwardedRef={[Function]}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "center",
                        "backgroundColor": "#5cb85c",
                        "borderBottomWidth": null,
                        "borderColor": "#007aff",
                        "borderLeftWidth": null,
                        "borderRadius": 50,
                        "borderRightWidth": null,
                        "borderTopWidth": null,
                        "borderWidth": undefined,
                        "elevation": 2,
                        "flex": 1,
                        "flexDirection": "row",
                        "height": 45,
                        "justifyContent": "space-between",
                        "marginHorizontal": "5%",
                        "opacity": 1,
                        "paddingBottom": 6,
                        "paddingTop": 6,
                        "shadowColor": undefined,
                        "shadowOffset": undefined,
                        "shadowOpacity": undefined,
                        "shadowRadius": undefined,
                      }
                    }
                  >
                    <Text
                      style={
                        Array [
                          Object {
                            "backgroundColor": "transparent",
                            "color": "#fff",
                            "fontFamily": "System",
                            "fontSize": 16.5,
                            "marginLeft": 0,
                            "marginRight": 0,
                            "paddingLeft": 16,
                            "paddingRight": 16,
                          },
                          Object {
                            "flex": 1,
                            "textAlign": "center",
                          },
                          Object {
                            "color": "black",
                          },
                        ]
                      }
                      uppercase={false}
                    >
                      Confirm
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </Modal>
        </View>
      </RCTScrollView>
    </RCTSafeAreaView>
  </View>
</View>
`;
